hack/lib/golang.sh

```diff
@@ -524,6 +524,15 @@ kube::golang::internal::verify_go_version() {
   if [ "${GOTOOLCHAIN:-auto}" != 'auto' ]; then
     # no-op, just respect GOTOOLCHAIN
     :
+  elif [ "${GO_VERSION:-}" == 'devel' ]; then
```

Comment:
I was sort of hoping we could move towards the GOTOOLCHAIN standard, we have a lot of ways to configure this currently, and it would be nice to rely on go tooling instead of shell/gimmee when possible.

Reply:
Does https://pkg.go.dev/golang.org/dl/gotip make sense?

Reply:
possibly, i was trying to stay within the current boundaries and not introduce more things.

Reply:
Done! https://github.com/kubernetes/kubernetes/issues/132204

Reply:
i saw `"devel"` in go repo - https://github.com/search?q=repo%3Agolang%2Fgo%20%2F%22devel%22%2F&type=code